bin/onlyswaps-solver/src/main.rs

```diff
@@ -86,9 +98,60 @@ async fn run(
             println!("received ctrl+c, shutting down...");
             Ok(())
         },
+    };
+
+    if let Some(arc_manager) = maybe_manager
+        && let Some(manager) = Arc::into_inner(arc_manager)
+    {
+        // ignore stop errors
+        match tokio::time::timeout(std::time::Duration::from_secs(1), manager.stop()).await {
+            Ok(Ok(_)) => (),
+            Ok(Err(e)) => tracing::error!(error = ?e, "Error while stopping omnievent manager"),
+            Err(_) => tracing::error!("Failed to stop omnievent within 1s timeout"),
+        }
     }
+
+    out
 }
 
 async fn setup(networks: HashMap<u64, Network<DynProvider>>) -> anyhow::Result<()> {
     setup_allowances(&networks).await
 }
+
+type ArcManager = Arc<EventManager<MultiProvider<u64>, NopDatabase>>;
+
+async fn get_omnievent_service(
+    maybe_endpoint: Option<
+        impl TryInto<Endpoint, Error: std::error::Error + Send + Sync + 'static>,
+    >,
+    networks: &HashMap<u64, Network<DynProvider>>,
+) -> anyhow::Result<(OmniEventBoxService, Option<ArcManager>)> {
+    if let Some(endpoint) = maybe_endpoint {
+        // Yay, we have an omnievent instance running already
+        let endpoint: Endpoint = endpoint.try_into()?;
+        let service = endpoint.connect().await?;
+
+        return Ok((service.map_err(Into::into).boxed(), None));
```

Comment:
ahh - this is when using external omnievent right? took a min to understand what was going on, so would add a comment to make it obvious

Reply:
yeah, exactly - will update